### PR TITLE
added first iteration of schedule update

### PR DESCRIPTION
**Pull request by axsaucedo**
# Schedule Update
## Details

This commit contains functionality to enable schedules to be updated with the latest status as opposed to  inserting a new schedule every time
# Tests

There are currently no tests. I am currently running the code and it has not crashed... _we will need tests soon_

### CODE REVIEW

File: models/__init__.py [hunk 1]
```diff
@@ -137,7 +137,7 @@ class Association(DarwinModel):
     associated_service  = ForeignKeyField(AssociationService, related_name="associated_service_set")
     tiploc              = CharField()
     category            = CharField(choices=ASSOCIATION_TYPE)
-    cancelled           = BooleanField()
+    cancelled           = BooleanField(null=True, default=False)
```
Comment (vladvelici) [inline models/__init__.py]: Why is cancelled nullable and not just False by default?

**Comment by vladvelici** [comment]
This code should work fine. I would do the following changes:

- add an unique index on the Schedule for the (`uid`, `rid`) pair,
- change the query from the complicated select() and then the count to a simpler `Schedule.get((Schedule.uid==uid) & (Schedule.rid == rid))`
- write tests as you mentioned (I'm writing a simple test now)

File: models/__init__.py (hunk 1 above)
Comment (axsauze) [inline models/__init__.py]: I might be wrong but I think default doesn't make it null able by default, or at least I had this error last time.

Comment (vladvelici) [inline models/__init__.py]: I'm just wondering why it needs to be nullable?

**Comment by axsauze** [comment]
Would the get throw an error if it finds more than one element? Also although I'm keen to start writing tests let's first make sure we have a proof of concept ready, as we want to hack something up quick before we establish ourselves and start writing tests

**Comment by vladvelici** [comment]
A simple test is the simplest way to see if it actually works, not thinking of writing tons of edge cases.
